server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -117,6 +119,11 @@ public class Indexer {
     private final Function<ColumnIdent, Reference> getRef;
     private final boolean writeOids;
     private final Version tableVersionCreated;
+    @Nullable
+    private final UpdateToInsert updateToInsert;
+    private final List<Reference> assignedColumns;
+    private final Supplier<List<Reference>> indexingOrder;
```

Comment:
Why do you need a supplier here? columns can be changed inbetween?

Reply:
Yes, the supplier is there to account for dynamically added columns.

Reply:
I believe though that for example the `IndexerTest#test_index_nested_array()` suffers from this.
the `indexingOrder` doesn't contain the `y` column which is dymanically added.

Reply:
Sorry it is because the commit is still a draft - 
```
this.indexingOrder = () -> ((DocTableInfo) (nodeCtx.schemas() == null ? table : nodeCtx.schemas().getTableInfo(table.ident()))).rootColumns();
```
And the indexer tests are missing `schemas` to be able to get the updated order, the integration tests should pass.

Reply:
ah, I see, thx!